presto-native-execution/presto_cpp/main/PeriodicServiceInventoryManager.h

```diff
@@ -33,6 +33,11 @@ class PeriodicServiceInventoryManager {
 
   void stop();
 
+  /// If disabled then won't send any requests, but keeps itself running.
+  void enableRequest(bool enable) {
+    requestEnabled_ = enable;
```

Comment:
Shall we add a log whenever requestEnabled_ is changed?
```
if (requestEnabled_.exchange(enable) != enable) {
   LOG(INFO) << enable or disable?
}
```

Reply:
We do this in PrestoServer::enableAnnouncer(), which is on a slightly higher level than this base class.